server/schedule/labeler/labeler.go

```diff
@@ -96,46 +96,75 @@ func (l *RegionLabeler) adjustRule(rule *LabelRule) error {
 
 	switch rule.RuleType {
 	case KeyRange:
-		data, ok := rule.Rule.(map[string]interface{})
-		if !ok {
-			return errs.ErrRegionRuleContent.FastGenByArgs(fmt.Sprintf("invalid rule type: %T", reflect.TypeOf(rule.Rule)))
-		}
-		startKey, ok := data["start_key"].(string)
-		if !ok {
-			return errs.ErrRegionRuleContent.FastGenByArgs(fmt.Sprintf("invalid startKey type: %T", reflect.TypeOf(data["start_key"])))
-		}
-		endKey, ok := data["end_key"].(string)
-		if !ok {
-			return errs.ErrRegionRuleContent.FastGenByArgs(fmt.Sprintf("invalid endKey type: %T", reflect.TypeOf(data["end_key"])))
-		}
-		var r KeyRangeRule
-		r.StartKeyHex, r.EndKeyHex = startKey, endKey
-		var err error
-		r.StartKey, err = hex.DecodeString(r.StartKeyHex)
+		r, err := l.adjustKeyRangeRule(rule.Rule)
 		if err != nil {
-			return errs.ErrHexDecodingString.FastGenByArgs(r.StartKeyHex)
+			return err
 		}
-		r.EndKey, err = hex.DecodeString(r.EndKeyHex)
-		if err != nil {
-			return errs.ErrHexDecodingString.FastGenByArgs(r.EndKeyHex)
+		rule.Rule = r
+		return nil
+	case KeyRanges:
+		rules, ok := rule.Rule.([]map[string]interface{})
+		if !ok {
+			return errs.ErrRegionRuleContent.FastGenByArgs(fmt.Sprintf("invalid rule type: %T", reflect.TypeOf(rule.Rule)))
 		}
-		if len(r.EndKey) > 0 && bytes.Compare(r.EndKey, r.StartKey) <= 0 {
-			return errs.ErrRegionRuleContent.FastGenByArgs("endKey should be greater than startKey")
+		rs := make([]*KeyRangeRule, 0, len(rules))
+		for _, r := range rules {
+			rr, err := l.adjustKeyRangeRule(r)
+			if err != nil {
+				return err
+			}
+			rs = append(rs, rr)
 		}
-		rule.Rule = &r
+		rule.Rule = rs
```

Comment:
How about using another name for `rule.Rule`? See https://github.com/pingcap/tidb/pull/27660#pullrequestreview-753499559

Reply:
I'm not quite undertand. Would you like to provide a better name?

Reply:
I have no idea about it =。=